zipkin-storage/elasticsearch/src/main/java/zipkin/storage/elasticsearch/InternalElasticsearchClient.java

```diff
@@ -55,17 +86,18 @@
    *
    * Will never return results; this interface is for aggregation only.
    */
-  ListenableFuture<Buckets> scanTraces(String[] indices, QueryBuilder query,
+  protected abstract ListenableFuture<Buckets> scanTraces(String[] indices, QueryBuilder query,
       AbstractAggregationBuilder... aggregations);
 
-  interface Buckets {
+  protected interface Buckets {
     /**
      * Retrieves the aggregation bucket keys from the aggregation at name[>nestedPath]
      *
      * The elasticsearch store only uses this one attribute of any aggregations, and so this
      * interface wraps the ugliness of dealing with different aggregation APIs while leaving control
      * of the aggregation keys in the hands of the caller.
      */
+    // TODO(adrian): revisit once the other implementation is in: can we avoid varags?
```

Comment:
We totally can – the varargs is just there to implement one-or-many semantics. Either this is a first-level aggregation, or it's nested "arbitrarily" deep (in practice, we only have two levels). This seemed clearer from both the implementation and user side to me than two methods (e.g. `getKeys` and `getNestedKeys`) .